From 1c16bcff9cbf115f0bed8c5296675b0188a1ea40 Mon Sep 17 00:00:00 2001
From: Kunhui-Li <kunhuix.li@intel.com>
Date: Fri, 20 Aug 2021 11:20:36 +0800
Subject: [PATCH 15/15] config_tools: add a new entry MAX_EFI_MMAP_ENTRIES

It is used to specify the maximum number of EFI memmap entries.

On some platforms, like Tiger Lake, the number of EFI memmap entries
becomes 268 when the BIOS settings are changed.
The current value of MAX_EFI_MMAP_ENTRIES (256) defined in hypervisor
is not big enough to cover such cases.

As the number of EFI memmap entries depends on the platforms and the
BIOS settings, this patch introduces a new entry MAX_EFI_MMAP_ENTRIES
in configurations so that it can be adjusted for different cases.

Tracked-On: #6442

Signed-off-by: Kunhui-Li <kunhuix.li@intel.com>
---
 misc/config_tools/data/adl-s-crb/hybrid.xml            | 1 +
 misc/config_tools/data/adl-s-crb/hybrid_rt.xml         | 1 +
 misc/config_tools/data/adl-s-crb/industry.xml          | 1 +
 misc/config_tools/data/adl-s-crb/logical_partition.xml | 1 +
 misc/config_tools/data/icx-rvp/industry.xml            | 1 +
 5 files changed, 5 insertions(+)

diff --git a/misc/config_tools/data/adl-s-crb/hybrid.xml b/misc/config_tools/data/adl-s-crb/hybrid.xml
index 8728d1c32..31f7ab1ca 100644
--- a/misc/config_tools/data/adl-s-crb/hybrid.xml
+++ b/misc/config_tools/data/adl-s-crb/hybrid.xml
@@ -49,6 +49,7 @@
         </MEMORY>
         <CAPACITIES>
             <IOMMU_BUS_NUM>0x100</IOMMU_BUS_NUM>
+            <MAX_EFI_MMAP_ENTRIES>256</MAX_EFI_MMAP_ENTRIES>
             <MAX_IR_ENTRIES>256</MAX_IR_ENTRIES>
             <MAX_IOAPIC_NUM>1</MAX_IOAPIC_NUM>
             <MAX_PCI_DEV_NUM>96</MAX_PCI_DEV_NUM>
diff --git a/misc/config_tools/data/adl-s-crb/hybrid_rt.xml b/misc/config_tools/data/adl-s-crb/hybrid_rt.xml
index 60a5a120a..e9ef5694a 100644
--- a/misc/config_tools/data/adl-s-crb/hybrid_rt.xml
+++ b/misc/config_tools/data/adl-s-crb/hybrid_rt.xml
@@ -48,6 +48,7 @@
     </MEMORY>
     <CAPACITIES>
       <IOMMU_BUS_NUM>0x100</IOMMU_BUS_NUM>
+      <MAX_EFI_MMAP_ENTRIES>256</MAX_EFI_MMAP_ENTRIES>
       <MAX_IR_ENTRIES>256</MAX_IR_ENTRIES>
       <MAX_IOAPIC_NUM>1</MAX_IOAPIC_NUM>
       <MAX_PCI_DEV_NUM>96</MAX_PCI_DEV_NUM>
diff --git a/misc/config_tools/data/adl-s-crb/industry.xml b/misc/config_tools/data/adl-s-crb/industry.xml
index 4a47d3bd2..2964c446e 100644
--- a/misc/config_tools/data/adl-s-crb/industry.xml
+++ b/misc/config_tools/data/adl-s-crb/industry.xml
@@ -51,6 +51,7 @@
     </MEMORY>
     <CAPACITIES>
       <IOMMU_BUS_NUM>0x100</IOMMU_BUS_NUM>
+      <MAX_EFI_MMAP_ENTRIES>256</MAX_EFI_MMAP_ENTRIES>
       <MAX_IR_ENTRIES>256</MAX_IR_ENTRIES>
       <MAX_IOAPIC_NUM>1</MAX_IOAPIC_NUM>
       <MAX_PCI_DEV_NUM>96</MAX_PCI_DEV_NUM>
diff --git a/misc/config_tools/data/adl-s-crb/logical_partition.xml b/misc/config_tools/data/adl-s-crb/logical_partition.xml
index 2f75d1e09..d6c463d9b 100644
--- a/misc/config_tools/data/adl-s-crb/logical_partition.xml
+++ b/misc/config_tools/data/adl-s-crb/logical_partition.xml
@@ -48,6 +48,7 @@
     </MEMORY>
     <CAPACITIES>
       <IOMMU_BUS_NUM>0x100</IOMMU_BUS_NUM>
+      <MAX_EFI_MMAP_ENTRIES>256</MAX_EFI_MMAP_ENTRIES>
       <MAX_IR_ENTRIES>256</MAX_IR_ENTRIES>
       <MAX_IOAPIC_NUM>1</MAX_IOAPIC_NUM>
       <MAX_PCI_DEV_NUM>96</MAX_PCI_DEV_NUM>
diff --git a/misc/config_tools/data/icx-rvp/industry.xml b/misc/config_tools/data/icx-rvp/industry.xml
index 7b6eebc4e..223d1c51e 100644
--- a/misc/config_tools/data/icx-rvp/industry.xml
+++ b/misc/config_tools/data/icx-rvp/industry.xml
@@ -51,6 +51,7 @@
     </MEMORY>
     <CAPACITIES>
       <IOMMU_BUS_NUM>0x100</IOMMU_BUS_NUM>
+      <MAX_EFI_MMAP_ENTRIES>256</MAX_EFI_MMAP_ENTRIES>
       <MAX_IR_ENTRIES>256</MAX_IR_ENTRIES>
       <MAX_IOAPIC_NUM>1</MAX_IOAPIC_NUM>
       <MAX_PCI_DEV_NUM>256</MAX_PCI_DEV_NUM>
-- 
2.25.1

